src/java/htsjdk/samtools/metrics/MetricsFile.java

```diff
@@ -267,10 +274,9 @@ private void printHistogram(final BufferedWriter out, final FormatUtil formatter
 
             for (final Histogram<HKEY> histo : nonEmptyHistograms) {
                 final Histogram<HKEY>.Bin bin = histo.get(key);
-                final double value = (bin == null ? 0 : bin.getValue());
-
                 out.append(SEPARATOR);
-                out.append(formatter.format(value));
+
+                if(bin != null) out.append(formatter.format(bin.getValue()));
             }
```

Comment:
These changes were added because there was an issue I ran into with the testing format where the 'writeThenReadBack' function was not passing the tests with two differently sized histograms because this was filling in the empty rows with 0s. The original MetricsFile object didn't have these 0s in the histograms but the new file did, making them unequal and causing tests to fail.